python/hsml/model.py

```diff
@@ -291,7 +296,27 @@ def get_feature_view(self):
             `Exception` in case the backend fails to retrieve the tags.
         """
         fv_prov = self.get_feature_view_provenance()
-        return explicit_provenance.Links.get_one_accessible_parent(fv_prov)
+        fv = explicit_provenance.Links.get_one_accessible_parent(fv_prov)
+        if fv is None:
+            return None
+        if init:
+            td_prov = self.get_training_dataset_provenance()
+            td = explicit_provenance.Links.get_one_accessible_parent(td_prov)
+            is_deployment = "DEPLOYMENT_NAME" in os.environ
+            if online or is_deployment:
+                _logger.info(
+                    "Initializing serving"
+                    + (" - deployment detected" if is_deployment else "")
+                )
+                fv.init_serving(training_dataset_version=td.version)
+            elif online is False:
+                _logger.info("Initializing batch")
+                fv.init_batch_scoring(training_dataset_version=td.version)
+            else:
+                _logger.info("Initializing is not configured")
+        else:
+            _logger.info("Initializing is turned off")
```

Comment:
As for the logs, IMHO:
- only the "initializing serving" is useful.
- "not configured" and "turned off" are the default behavior (i.e., not calling fv.init...) so they are not needed.
- "Initializing batch" can be confusing for users. Maybe something like "Initializing for batch retrieval" is more informative. Or "Initializing for batch retrieval of feature vectors".
- In the same way, maybe "Initializing for batch and online retrieval of feature vectors" is more descriptive than "Initializing serving".

Reply:
done